### ISSUE TITLE
Insert defaultValue is not typed if field is a nullable or optional array

### ISSUE THREAD

**Issue by nianiam** [edmundhung/conform#514]
### Describe the bug and the expected behavior

I am creating some dynamic fields in my form using the `form.insert` method, but noticed that I was getting a type error  whilst trying to add some default values. It would appear that when the field is an array and is optional/nullable then the `defaultValue` type becomes a string. Here is a simplified example

```ts
export const schema = z.object({
  test: z.array(z.object({ nested: z.string() })).nullable() // or .optional()
})
```

```ts
<button
  {...form.insert.getButtonProps({
    name: fields.test.name,
    defaultValue: { // <-- Type '{ nested: string; }' is not assignable to type 'string'.
      nested: "I am nested",
    },
  })}
>
  Click me
</button>
```

It's fairly easy to work aroud this situation by making the field required and setting the default value of the form to an empty array, however, I thought I would report in case this was not deliberate!

Maintaining the type inside of the array would make the most sense.

### Conform version

v1.0.3

### Steps to Reproduce the Bug or Issue

1. Create a simple schema with an an optional/nullable array as a field.
2. Use the field metadata to insert values in the array.
3. See defaultValue is types as a string and not the type within the schema.

### What browsers are you seeing the problem on?

Chrome, Firefox, Safari

**Comment by lifeiscontent** [comment]
@edmundhung this seems to be related to the defaultValue fix that @aaronadamsCA did whenever a schema has a key that is an array that also has an `optional()` or `nullable()` this seems to still have an issue.

**Comment by edmundhung** [comment]
Thanks for reporting the issue! This is fixed in [v1.1.3](https://github.com/edmundhung/conform/releases/tag/v1.1.3).
